packages/core-data/src/hooks/use-query-select.ts

```diff
@@ -97,19 +98,31 @@ const enrichSelectors = memoize( ( selectors ) => {
 		}
 		Object.defineProperty( resolvers, selectorName, {
 			get: () => ( ...args ) => {
-				const {
-					getIsResolving,
-					hasStartedResolution,
-					hasFinishedResolution,
-				} = selectors;
+				const { getIsResolving, hasFinishedResolution } = selectors;
 				const isResolving = !! getIsResolving( selectorName, args );
+				const hasResolved =
+					! isResolving &&
+					hasFinishedResolution( selectorName, args );
+				const data = selectors[ selectorName ]( ...args );
+
+				let status;
+				if ( isResolving ) {
+					status = Status.Resolving;
+				} else if ( hasResolved ) {
+					if ( data ) {
+						status = Status.Success;
```

Comment:
even though we didn't introduce the constants in this PR it would be nice if we used a `const enum` instead of an `enum` so we don't have to carry around the object-lookup and associated baggage. oh well.